### PR TITLE
Add repl support to binary and test targets

### PR DESCRIPTION
**Pull request by johnynek**
actually, starting repls for test code and binaries is really useful, so I added support for that as well. This is done by making the binary targets return their deps correctly.

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by johnynek** [comment]
@dinowernli @smparkes @colinmarc @ianoc ?

File: scala/scala.bzl [hunk 1]
```diff
@@ -368,14 +368,23 @@ def _scala_macro_library_impl(ctx):
 # Common code shared by all scala binary implementations.
 def _scala_binary_common(ctx, cjars, rjars):
   write_manifest(ctx)
-  _compile_or_empty(ctx, cjars, [], False)  # no need to build an ijar for an executable
+  outputs = _compile_or_empty(ctx, cjars, [], False)  # no need to build an ijar for an executable
```
Comment (ianoc) [inline scala/scala.bzl]: should we give this a different name so its separate from the outputs defined on 379?

**Comment by ianoc** [comment]
looks fine to me

**Comment by dinowernli** [comment]
LGTM

**Pull request merged**